common.sh

```diff
@@ -35,9 +35,29 @@ delete_pg()
     if [ ${ENABLE_PLACEMENT_GROUP} -eq 0 ] || [ -z $PLACEMENT_GROUP ]; then
         return 0
     fi
-    AWS_DEFAULT_REGION=us-west-2 aws ec2 delete-placement-group \
-        --group-name ${PLACEMENT_GROUP}
-    return $?
+    local ret=0
+    local retry=20
+    while [ "$retry" -ge 0 ]; do
+        AWS_DEFAULT_REGION=us-west-2 aws ec2 delete-placement-group \
+            --group-name ${PLACEMENT_GROUP}
+        ret=$?
+        # Error code 255 is InvalidPlacementGroup.InUse, which means the
+        # The placement group is in use and may not be deleted, due to
+        # the attached ec2 instances are not terminated completely.
+        if [ "$ret" -eq 255 ]; then
+            echo "The placement group is in use and may not be deleted, retry after 10 seconds."
+            sleep 10
+        else
+            break
+        fi
+        retry=$((retry-1))
+    done
+    if [ "$ret" -ne 0 ]; then
+        echo "Fail to delete placement group."
```

Comment:
do we log the actual placement group name/id somewhere?

Reply:
Thanks, added the placement group name in the echo messages in latest revision.